include/swift/AST/Attr.def

```diff
@@ -397,7 +397,7 @@ DECL_ATTR(_private, PrivateImport,
 
 // SWIFT_ENABLE_TENSORFLOW
 DECL_ATTR(differentiable, Differentiable,
-  OnAccessor | OnConstructor | OnFunc | OnVar | LongAttribute |
+  OnAccessor | OnConstructor | OnFunc | OnVar | OnSubscript | LongAttribute |
```

Comment:
Relatedly, if we do static callable as `call` keyword, we'd need an `OnCall` kind.

Reply:
Should we support some of these declaration types for `@differentiating` too?
`OnVar` would be very weird. `OnAccessor` and `OnConstructor` would be kind of weird.

Reply:
I'm torn about this, and would like to defer this to evolution reviews. Let's start with only allowing `@differentiating` on real function declarations.

Reply:
SGTM.